core/gather/gatherers/link-elements.js

```diff
@@ -116,6 +116,9 @@ class LinkElements extends FRGatherer {
    * @return {Promise<LH.Artifacts['LinkElements']>}
    */
   static async getLinkElementsInHeaders(context, devtoolsLog) {
+    // We currently cannot get the main resource in timespan mode.
+    if (context.gatherMode === 'timespan') return [];
```

Comment:
It depends on the timespan, though.

We may need to audit our usage of `LinkElements` to double check if we assume they will always contain all the LinkElements of the page. I assume we have always assumed that's true...

Reply:
> It depends on the timespan, though.

Are you saying we could look through the network records and find *all* main document requests over a period of time and use that?

> Maybe it's better for the gatherer to throw for now?

If the gatherer throws then we will always get an error in timespan mode (with the changes in this PR).

Reply:
allowing it to throw breaks two fairly involved api tests, so it's not a very simple approach

Reply:
throwing is better than just making LinkElements not support timespan?

pretty sure this is the first time LinkElements has ever been run for timespan, otherwise wouldn't this have error from the sample flow json long ago? seems fine to do this for now. For the purpose of favicons, only in-page link elements can define an `icon`. headers cannot

Reply:
> Are you saying we could look through the network records and find all main document requests over a period of time and use that?

No, just talking about what would work right now. If the timespan period included the final mainDocumentUrl's request, we can get the main resource request.

I guess at least returning `[]` here is consistent, you just might want to document that wherever we record the differences in artifacts based on gatherer mode.

> seems fine to do this for now

There aren't any LR users running timespans anyways, so it doesn't seem like it's fixing anything for timespans either?